tests/integration/record_benchmark.py

```diff
@@ -46,55 +51,61 @@
 
 data = {}
 
+cloudwatch_report_schema = {
+    "totalTimeMills": 'Milliseconds',
+    "totalRequests": 'Count',
+    "failedRequests": 'Count',
+    "concurrentClients": 'Count',
+    "totalTokens": 'Count',
+    "tokenPerRequest": 'Count',
+    "averageLatency": 'Milliseconds',
+    "p50Latency": 'Milliseconds',
+    "p90Latency": 'Milliseconds',
+    "p99Latency": 'Milliseconds',
+    "timeToFirstByte": 'Milliseconds',
+    "p50TimeToFirstByte": 'Milliseconds',
+    "p90TimeToFirstByte": 'Milliseconds',
+    "p99TimeToFirstByte": 'Milliseconds',
+}
+
 
 class Benchmark:
 
-    def __init__(self, dyn_resource):
+    def __init__(self, dyn_resource, data: dict):
         self.dyn_resource = dyn_resource
         self.table = dyn_resource.Table("RubikonBenchmarks")
         self.table.load()
+        self.data = data
 
     def add_benchmark(self):
-        self.table.put_item(Item=data)
+        self.table.put_item(Item=self.data)
 
 
-def record_table():
+def record_table(data: dict):
     table = boto3.resource("dynamodb").Table("RubikonBenchmarks")
     table.put_item(Item=data)
 
 
-def record_cloudwatch():
+def record_cloudwatch(data: dict):
     esc = lambda n: n.replace("/", "-").replace(".", "-").replace("=", "-"
                                                                   ).strip(' -')
     job_name = data["modelId"] if "job" not in data else data["job"]
-    metric_name = lambda n: f"lmi_{data['instance']}_{esc(data['image'])}_{esc(job_name)}_{n}"
-    metric_data = [
-        {
-            'MetricName': metric_name("throughput"),
-            'Unit': 'Count/Second',
-            'Value': data['throughput']
-        },
-        {
-            'MetricName': metric_name("latency_p50"),
-            'Unit': 'Milliseconds',
-            'Value': data['P50']
-        },
-        {
-            'MetricName': metric_name("latency_p90"),
-            'Unit': 'Milliseconds',
-            'Value': data['P90']
-        },
-        {
-            'MetricName': metric_name("latency_p99"),
-            'Unit': 'Milliseconds',
-            'Value': data['P99']
-        },
-    ]
+    benchmark_vars = data["benchmark_vars"] if data["benchmark_vars"] else ""
+    metric_name = lambda n: (f"lmi_{data['instance']}_{esc(data['image'])}"
+                             f"_{esc(job_name)}_{esc(benchmark_vars)}_{n}")
+    metric_data = []
+    for metric, unit in cloudwatch_report_schema.items():
+        if metric in data.keys():
+            metric_data.append({
+                'MetricName': metric_name(metric),
```

Comment:
There is some fudging of the metric names here. We have names in the cloudwatch_report_schema, json produced by awscurl, data object, and log lines parsed in `data_from_client` from `benchmark.log`. These might differ, but we should try to get them to all match. Right now, the `benchmark.log` will differ from the cloudwatch upload and the awscurl ones, while the data object could have either

Maybe the quickest is to make the names from `benchmark.log` match the ones in the cloudwatch_report_schema/awscurl. This should align all the names. If you do this, you will also need to update the tables in the cloudwatch dashboard as the metrics they are displaying are being renamed